src/lifecycles/lifecycle.helpers.ts

```diff
@@ -149,10 +137,7 @@ export interface InternalParcel {
   timeouts: AppOrParcelTimeouts;
 }
 
-export type AppOrParcel = ApplicationObject | Parcel;
-
-export type ApplicationObject<ExtraProps = {}> = LifeCycles<ExtraProps> &
```

Comment:
Replaced by `InternalApplication` type which is a more accurate representation of the type